### ISSUE TITLE
Multiple Projects Open In Single Window Bug

### ISSUE THREAD

**Issue by RichardTMiles** [cunla/ghactions-toolbar#37]
Thank you for making this plugin!!!!! You made my day here.

**Describe the bug**

When IntelliJ is set up to open multiple projects in a single window using tabs, the plugin only shows information for the first `repositories` actions.

**Describe the bug:**
Multiple Projects Open In Single Window Bug

**Steps to reproduce:** 
Open multiple projects that are actions enabled in IntelliJ's tab mode.

**Expected behavior:**
The plugin should be responsive to what tab (project) is selected.

**Comment by RichardTMiles** [comment]
ooooh, nvm I'd also love to see log support for over 1mb.

**Comment by cunla** [comment]
Which version are you using?

**Comment by cunla** [comment]
- It should show all repositories in the workspace - if it does not, please give more details...
- ConsoleView in intellij supports 1mb, I changed the behavior so there is a console per job, i.e., the log is 1mb per job instead of per workflow - that should suffice. More than 1mb would not support the log folding properly.

**Comment by RichardTMiles** [comment]
I will re-install from intelliJ's marketplace plugins menu and give her another try today. Is that the best way to get the updates?

**Comment by RichardTMiles** [comment]
<img width="1230" alt="Screen Shot 2022-09-28 at 12 47 45 PM" src="https://user-images.githubusercontent.com/9538357/192853148-38d68901-5618-4455-ad0d-0a7ddf87795d.png">

So I will log my experience from the beginning of a new install right now.  ^ off a fresh install with no intellij restart, we can see that no repositories were found. This was also my experience the first time. I should be prompted to restart intellij, or it just work :) 

<img width="1292" alt="Screen Shot 2022-09-28 at 12 52 21 PM" src="https://user-images.githubusercontent.com/9538357/192853875-d5594e4b-5767-4965-b8fa-2ba4aaa281ed.png">


Then I am still met with the issue.

**Comment by RichardTMiles** [comment]
I will be uninstalling :) Please let me know if I can help any other way.

**Comment by RichardTMiles** [comment]
I very much want this to work and will follow this thread!

**Comment by RichardTMiles** [comment]
<img width="927" alt="Screen Shot 2022-09-28 at 12 54 54 PM" src="https://user-images.githubusercontent.com/9538357/192854407-63c3c52a-8bfc-4671-9866-97a0d2704d7d.png">


Our logs can easily reach 5k lines

**Comment by cunla** [comment]
Let's separate it into two issues, one for the workflow runs and another for the log size.
- For the log size, the limit is not 1k lines - see my previous comment, 1mb (chars).
- For the workflow runs - can you please describe what git repositories are in the workspace and what do you see in the toolbar? it is not clear if you are seeing one repo or none.

**Comment by RichardTMiles** [comment]
ConsoleView in intellij supports 1mb - so you may need to lazy load content from memory to support larger logs.

For the workflow runs - I have two git repo's open, each in different windows (or tabs in this case). One is public and the other is private. https://github.com/RichardTMiles/CarbonPHP

Nothing particularly special about them

**Comment by RichardTMiles** [comment]
<img width="828" alt="Screen Shot 2022-10-04 at 9 00 25 PM" src="https://user-images.githubusercontent.com/9538357/193963245-1ba2073f-feaa-4af9-8dc9-4cb5085a003b.png">


com.intellij.util.IncorrectOperationException: Sorry but parent: com.dsoftware.ghtoolbar.workflow.data.WorkflowRunListLoader@1996f2be (class com.dsoftware.ghtoolbar.workflow.data.WorkflowRunListLoader) has already been disposed (see the cause for stacktrace) so the child: newDisposable (class com.intellij.openapi.util.Disposer$1) will never be disposed
	at com.intellij.openapi.util.ObjectTree.register(ObjectTree.java:53)
	at com.intellij.openapi.util.Disposer.register(Disposer.java:118)
	at com.dsoftware.ghtoolbar.workflow.data.WorkflowRunListLoader.reset(WorkflowRunListLoader.kt:40)
	at com.dsoftware.ghtoolbar.ui.wfpanel.WorkflowRunListLoaderPanel.displayEmptyStatus$lambda-6(WorkflowRunListLoaderPanel.kt:201)
	at com.intellij.util.ui.StatusText$1.onClick(StatusText.java:79)
	at com.intellij.ui.ClickListener$1.mouseReleased(ClickListener.java:59)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:298)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6654)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3345)
	at java.desktop/java.awt.Component.processEvent(Component.java:6419)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5029)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4861)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2793)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4861)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:898)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:820)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:743)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:439)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:803)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:438)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:119)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:604)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:436)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:873)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:484)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)

**Comment by RichardTMiles** [comment]
I would open multiple tickets, sure, but I feel this is going to be a deep hole (no offence). I'll poke my head in every few days to see how you're doing.

**Comment by RichardTMiles** [comment]
<img width="1016" alt="Screen Shot 2022-10-04 at 9 04 21 PM" src="https://user-images.githubusercontent.com/9538357/193963662-4671e4ed-1a27-49d4-ad26-4b5fc13ffc85.png">

Had to restart IntelliJ to get the above error to go away. This image captures 2 issues (our 1mb) and the WRONG git actions showing in this window. This is the actions for the private repo, not CarbonPHP as expected.

**Comment by RichardTMiles** [comment]
I recognize the new release is out, not sure why IntelliJ won't allow me to update... is it difficult to update their marketplace? 

I will update manually.

**Comment by RichardTMiles** [comment]
Updating intellij through the GUI didn't work, I suppose the patch files don't adjust the build version information thus I get the following 
<img width="437" alt="Screen Shot 2022-10-04 at 9 18 21 PM" src="https://user-images.githubusercontent.com/9538357/193965907-eb82ae95-b0df-4076-8b13-ce88d18ffcb0.png">

Just re-downloaded from online

**Comment by RichardTMiles** [comment]
<img width="340" alt="Screen Shot 2022-10-04 at 9 29 13 PM" src="https://user-images.githubusercontent.com/9538357/193966519-0b79cdd1-e642-4160-b4a6-d7b8ee92edec.png">

Directly downloading from online saw the plugin update automatically, but not through GUI!

**Comment by RichardTMiles** [comment]
Okay so now I'm on the latest release. After opening a single project and waiting for `waiting to index` stage to complete, I then opened the other object for the first time in this version and `Nothing to show` then appeared in the first project (after already loading successfully). I decided to wait for the second project to load completely to see if the fist would be affected after the index was completed but unfortunately this was not the case.



<img width="941" alt="Screen Shot 2022-10-04 at 9 38 38 PM" src="https://user-images.githubusercontent.com/9538357/193969137-2003542f-7fbb-4583-8885-af540922a7c7.png"
<img width="885" alt="Screen Shot 2022-10-04 at 9 38 47 PM" src="https://user-images.githubusercontent.com/9538357/193969160-8b9e8928-34ee-44e7-b484-85451418df4b.png">
>

**Comment by RichardTMiles** [comment]
https://stackoverflow.com/questions/65397429/how-to-open-projects-as-tabs-in-intellij

When I refer to tab view I'm using the following view feature:
<img width="605" alt="H7GlQ" src="https://user-images.githubusercontent.com/9538357/193970501-bbac3719-da4a-4206-82a9-06942b7a48cc.png">

**Comment by RichardTMiles** [comment]
Nothing to show proceeded to show in the price repo after multiple restarts; I then selected the wrench icon and unselected the following and hit apply.
<img width="307" alt="Screen Shot 2022-10-04 at 9 56 02 PM" src="https://user-images.githubusercontent.com/9538357/193971223-e2f199a5-8991-4b47-ae23-edd01f0486d0.png">


After doing so an infinite loading (or what appeared to be) showed up in the Github action tab. I then reselected the checkbox and that caused the plugin to start working again.
